### ISSUE TITLE
Exclude List

### ISSUE THREAD

**Issue by iceman73** [cmcconomy/iptv-filter#2]
Would you be open to adding an Exclude Categories Method?

**Comment by cmcconomy** [comment]
Currently, entering the secondary "category" screen has the effect of showing as selected, all categories that have at least one selected channel.

Clicking submit deselects every channel then selects every individual channel in every category that is checked as true. 

This is a leftover behavior from when the ui only engaged with categories but can still come in handy so í haven't deprecated it yet.

Can you elaborate on the use case or behavior your looking for exactly?

**Comment by iceman73** [comment]
My provider doesn't make an easy way to carve out VOD and TV Shows, so i wanted an ability to Filter out all TV channels and the remaining channels are the Movie and TV Show categories.

**Comment by cmcconomy** [comment]
that's precisely what encouraged me to write this m3u proxy.

Have you run it yet? By default no channels are selected, and you can search/ filter through channel or category names to select channels that the application includes in the outgoing m3u. This should allow you to choose only the channels you want

**Comment by iceman73** [comment]
Yes and it helped for the Movies as they are all under Category Movie.

As for TV Show's each Show is it own Category, and I didn't want to have select each new TV Show as they add it.

BTW I am very impressed and using it and love the ease of it.

**Comment by iceman73** [comment]
I am actually thinking of setting up multiple instances so i can do TV Channel filtering, Movies, and I may break out sports to its own list as well. Playing around with some other projects as well.

**Comment by cmcconomy** [comment]
Okay I think I understand what you're saying.. the m3u source file keeps changing as new shows are added, and you want to set up rules for whether to include it exclude new entries. Is that right?

**Comment by iceman73** [comment]
You got it

**Comment by cmcconomy** [comment]
I'll give it some thought thanks for the idea

**Comment by cmcconomy** [comment]
> You got it

If you upgrade to the latest version (instructions in README) I added a feature to help - any channels added since the last time you visited the filter screen are flagged as new.
This may eventually be leveraged to also provide the ability to whitelist/blacklist channel name strings to auto-include.. some other time :p

In the meantime I hope this helps.

**Comment by iceman73** [comment]
Working well. THis will help for time being. I may get some time to see if I could add some code to help with the allow/block lists

**Comment by cmcconomy** [comment]
Thanks. The idea I had (open to change) was every time we reload m3u, to look for all new items and call a method to determine whether to include, see:
https://github.com/cmcconomy/iptv-filter/blob/e1a02e641b598ff3508bbc932d24256b3b16771c/iptv_updater/iptv_updater.py#L96

then we need probably an admin screen to define field_name and regex_pattern (and whether whitelist/blacklist), as well as default behaviour (include new channels + blacklist, vs exclude new channels + whitelist).

- Craig
